src/scmdata/groupby.py

```diff
@@ -23,59 +36,71 @@ def _maybe_wrap_array(original, new_array):
         return new_array
 
 
-class _GroupBy(ImplementsArrayReduce):
```

Comment:
This diff is a bit hard to read, but I removed the `_GroupBy` class and added the functionality to `RunGroupBy`